### ISSUE TITLE
documentation: description of `setParseAction` is incorrect

### ISSUE THREAD

**Issue by mgrandi** [pyparsing/pyparsing#213]
in this section: https://pyparsing-docs.readthedocs.io/en/latest/HowToUsePyparsing.html#id6

the description for `setParseAction` is wrong:

```plaintext
setParseAction(*fn) - specify one or more functions to call after successful matching of the element; each function is defined as fn(s, loc, toks), where:

    s is the original parse string
    loc is the location in the string where matching started
    toks is the list of the matched tokens, packaged as a ParseResults object

Multiple functions can be attached to a ParserElement by specifying multiple arguments to setParseAction, or by calling setParseAction multiple times.
```

this is incorrect, `setParseAction` overwrites any previous parse action, while `addParseAction` appends the passed in callable to a list. Calling `setParseAction` multiple times only uses the last one , while calling `addParseAction` multiple times uses them all. 

I just got bitten by this and was wondering why it wasn't working so I decided to file a bug :)

see https://github.com/pyparsing/pyparsing/blob/master/pyparsing/core.py#L464-L473

and https://github.com/pyparsing/pyparsing/blob/master/pyparsing/core.py#L481-L483

**Comment by ptmcg** [comment]
Good catch, thanks!

**Comment by ptmcg** [comment]
This file needed some more extensive updating, thanks again.

**Comment by mgrandi** [comment]
looks like it solves this issue! i would maybe consider having this be generated from the docstrings of the relevant methods and have the documentation be auto generated, seems like it would just get out of date again in the future

**Comment by ptmcg** [comment]
Visit https://pyparsing-docs.readthedocs.io/en/latest/pyparsing.html to see the auto-generated docs from the pyparsing docstrings.

**Comment by ptmcg** [comment]
Released in 3.0.0a2
